src/poc/client/poc-client/src/app/(api)/_utils/wrapper.tsx

```diff
@@ -0,0 +1,51 @@
+const baseUrl="http://localhost:5001";
+  
+async function apiFetch<T>(endpoint: string, options: RequestInit = {}): Promise<T> {
+    const url = `${baseUrl}/${endpoint}`;
+  
+    const response = await fetch(url, {
```

Comment:
this should be wrapped in try catch block

Reply:
lines 6 to 19

Reply:
Most guides i saw say that the error is better caught when calling the async function (the throw error part is mainly to reject the promise). I changed the test page to add error handling.